packages/calcite-components/src/components/alert/alert.tsx

```diff
@@ -441,16 +449,18 @@ export class Alert implements OpenCloseComponent, LoadableComponent, T9nComponen
 
   @State() hasEndActions = false;
 
+  // todo: this queue stack is maintained as a state on every alert instance? Should we refactor this?
   /** the list of queued alerts */
   @State() queue: HTMLCalciteAlertElement[] = [];
```

Comment:
This is bad. We should refactor this. Currently, the stack is maintained as state on every component and it can get out of sync if multiple alerts are open by default. 

This should be some kind of `AlertManager` class to handle this and get rid of the internal events being emitted.

Reply:
related to https://github.com/Esri/calcite-design-system/issues/9340

Reply:
and https://github.com/Esri/calcite-design-system/issues/9499

Reply:
Opening PR to tackle after this one: #10032